### PR TITLE
Better Unknown Exception Handling

### PR DESCRIPTION
**Pull request by xenoterracide**
better handle unknown exceptions per #1200 and #1211, addresses the security vulnerability, and would provide needed information in the log to debug.

### CODE REVIEW

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/error/DefaultWebResponseExceptionTranslator.java [hunk 1]
```diff
@@ -111,8 +113,8 @@ public int getHttpErrorCode() {
 	@SuppressWarnings("serial")
 	private static class ServerErrorException extends OAuth2Exception {
 
-		public ServerErrorException(String msg, Throwable t) {
```
Comment (jgrandja) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/error/DefaultWebResponseExceptionTranslator.java]: No need to remove this constructor. Let's keep it. This needs to be a passive and binary compatible change.

Comment (xenoterracide) [reply spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/error/DefaultWebResponseExceptionTranslator.java]: given that the class itself is private is it necessary for binary compatibility? but sure I can make that change

Comment (xenoterracide) [reply spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/error/DefaultWebResponseExceptionTranslator.java]: also worth saying my constructor isn't public for any particular reason, just copying the code around it

**Comment by jgrandja** [comment]
Closing as per [comment](https://github.com/spring-projects/spring-security-oauth/issues/1200#issuecomment-345847497)

**Pull request closed without merging**